includes/Hooks.php

```diff
@@ -248,6 +250,52 @@ public static function onCodeEditorGetPageLanguage( $title, &$lang, $model, $for
 		return true;
 	}
 
+    /**
+     *  During the page move process, this updates the content model of the page to sanitize-css
+     * @param $old
+     * @param $new
+     * @param $userIdentity
+     * @param int $pageid
+     * @param int $redirid
+     * @param string $reason
+     * @param $revision
+     * @return true
+     */
+    public static function onPageMoveCompleting($old, $new, $userIdentity, int $pageid, int $redirid, string $reason, $revision) {
+        $config = self::getConfig();
+        $namespace = $new->getNamespace();
+        if ($revision->getContent( SlotRecord::MAIN )->getModel() === 'sanitized-css' ||
+            $namespace !== $config->get( 'TemplateStylesDefaultNamespace' ) || substr( $new->getText(), -4 ) !== '.css' ) {
+            return true;
+        }
+
+        $text = $revision->getContent( SlotRecord::MAIN )->getText();
+        $content = new TemplateStylesContent( $text );
+        $wikiPageFactory  = MediaWikiServices::getInstance()->getWikiPageFactory();
+        $userFactory  = MediaWikiServices::getInstance()->getUserFactory();
+        $titleDBKey = $new->getDBkey();
+
+        $title = Title::newFromText( $titleDBKey, $config->get( 'TemplateStylesDefaultNamespace' ) );
+        $page = $wikiPageFactory->newFromTitle( $title );
+
+        if ( !$page->exists() ) {
+            return true;
+        }
+
+        $user = $userFactory->newFromId( $userIdentity->getId() );
+        $summary = CommentStoreComment::newUnsavedComment( trim( 'On Page Move Completing Content Model Change' ) );
```

Comment:
This should be prettier, it will be visible on page history for everyone. We'd also need translations, which complicates things a bit and I'm wondering if just leaving it empty isn't a better way to solve this